tests/test_bugzilla.py

```diff
@@ -0,0 +1,498 @@
+    """
+
+    def __init__(self, status_code=200, params=None, text=TEXT):
+        self.status_code = status_code
+        self.params = params
+        self.text = text
+
+    def get(self, url):
+        if '/rest/product/' in url:
+            return FakeResponseProject(status_code=self.status_code, text=self.text)
+        else:
+            return FakeResponse(status_code=self.status_code, text=self.text)
+
+    def post(self, url, json, headers='headers'):
+        if self.status_code == 204:
+            return FakeResponseID(status_code=self.status_code, text=self.text)
+        else:
+            return FakeResponse(status_code=self.status_code, text=self.text)
+
+    def put(self, url, json):
+        return FakeResponse(status_code=self.status_code, text=self.text)
+
+    def close(self):
+        return
+
+
+class FakeResponse(object):
+    """
+    Mock response coming from server via Requests.
+    """
+
+    def __init__(self, status_code=200, text=TEXT):
+        self.status_code = status_code
+        self.text = text
+
+    def raise_for_status(self):
+        if self.status_code not in [200, 204, 401, 402]:
+            raise requests.RequestException
+
+    def json(self):
+        if self.status_code == 200:
+            return {}
+        elif self.status_code == 401:
+            return {'error': True, 'message': 'There is some error.'}
+        elif self.status_code == 402:
+            return {'bugs': {0: {'changes': {}}}}
+
+
+class FakeResponseProject(FakeResponse):
+
+    def json(self):
+        if self.status_code == 204:
+            return {"products": []}
+        elif self.status_code == 200:
+            return {}
+
+
+class FakeResponseID(FakeResponse):
+
+    def json(self):
+        if self.status_code == 204:
+            return {'id': TICKET_ID2}
+
+
+class TestBugzillaTicket(TestCase):
+
+    @patch.object(bugzilla.BugzillaTicket, '_create_requests_session')
+    def test_generate_ticket_url(self, mock_session):
+        mock_session.return_value = FakeSession()
+        ticket = bugzilla.BugzillaTicket(URL, PROJECT, ticket_id=TICKET_ID)
+        self.assertEqual(TICKET_URL, ticket._generate_ticket_url())
+        self.assertEqual(ticket.request_result, SUCCESS_RESULT._replace(url=TICKET_URL))
+
+    @patch.object(bugzilla.BugzillaTicket, '_create_requests_session')
+    def test_generate_ticket_url_no_ticket_id(self, mock_session):
+        mock_session.return_value = FakeSession()
+        ticket = bugzilla.BugzillaTicket(URL, PROJECT)
+        self.assertEqual(None, ticket._generate_ticket_url())
+        self.assertEqual(ticket.request_result, SUCCESS_RESULT)
+
+    @patch.object(ticketutil.ticket.Ticket, '_create_requests_session')
+    def test_create_requests_session_kerberos_auth(self, mock_session):
+        mock_session.return_value = FakeSession()
+        with patch.object(bugzilla.BugzillaTicket, '_create_requests_session'):
+            ticket = bugzilla.BugzillaTicket(URL, PROJECT)
```

Comment:
If you make the change in bugzilla.py below, you'll probably need to add the following to this line:

`auth='kerberos'`

Reply:
I am aware of that. I will wait for your response to the comment above.